include/paging.h

```diff
@@ -337,24 +337,54 @@ static inline PhysPt PAGING_GetPhysicalAddress(PhysPt linAddr) {
 
 /* Special inlined memory reading/writing */
 
-static inline uint8_t mem_readb_inline(PhysPt address) {
-	HostPt tlb_addr=get_tlb_read(address);
-	if (tlb_addr) return host_readb(tlb_addr+address);
-	else
+template <typename T>
+#if C_DEBUG && C_HEAVY_DEBUG
+// forwarded from debug
+void DEBUG_UpdateMemoryReadBreakpoints(const PhysPt addr);
+#else
+static void DEBUG_UpdateMemoryReadBreakpoints([[maybe_unused]] const PhysPt addr)
+{ /* no-op */
+}
+#endif
```

Comment:
It would be nice to co-locate the `#if C_DEBUG && C_HEAVY_DEBUG` branches out of the paging code and into `debug.h` header (where it's more applicable), and let the `debug.h` decide if it wants to create a placeholder functions (or not).

@LowLevelMahn , if you want to add another commit, I think we can use this for the full body of `include/debug.h`:

```c++
#ifndef DOSBOX_DEBUG_H
#define DOSBOX_DEBUG_H

#include "dosbox.h"
#include "mem.h"

#if C_DEBUG
void DEBUG_DrawScreen();
bool DEBUG_Breakpoint();
bool DEBUG_IntBreakpoint(uint8_t intNum);
void DEBUG_Enable(bool pressed);
void DEBUG_CheckExecuteBreakpoint(uint16_t seg, uint32_t off);
bool DEBUG_ExitLoop(void);
void DEBUG_RefreshPage(int scroll);
Bitu DEBUG_EnableDebugger();

extern Bitu cycle_count;
extern Bitu debugCallback;
#else // Empty debugging replacements
#endif

#if C_DEBUG && C_HEAVY_DEBUG
bool DEBUG_HeavyIsBreakpoint();
void DEBUG_HeavyWriteLogInstruction();
template <typename T>
void DEBUG_UpdateMemoryReadBreakpoints(const PhysPt addr);
#else // Empty heavy debugging replacements
template <typename T>
constexpr void DEBUG_UpdateMemoryReadBreakpoints(const PhysPt) { /* no-op */ }
#endif // C_DEBUG && C_HEAVY_DEBUG

#endif // DOSBOX_DEBUG_H
```

This will let us eliminate this `#if C_DEBUG && C_HEAVY_DEBUG` here, and we can simply `#include "debug.h"` at the top of `paging.h`.

Reply:
this way the code will not compile when C_DEBUG && C_HEAVY_DEBUG not active - because the DEBUG_UpdateMemoryReadBreakpoints-stub is then missing in paging.h - which re-introduces the #ifdefs